pymacaroons/caveat.py

```diff
@@ -15,7 +15,7 @@ def __init__(self,
 
     @property
     def caveat_id(self):
-        return convert_to_string(self._caveat_id)
+        return self._caveat_id
```

Comment:
I added backward compatibility for identifier and caveat_id, please have a look.